File: DonationSite.Core.Entities/Donate/Donate.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DonationSite.Core.Entities.Donate
{
    public class Donate
    {
        public int DonateID { get; set; }
        [Required]
        public string DonatorName { get; set; }
        public decimal Value { get; set; }
        [ForeignKey(nameof(Site))]
        public int FKSiteID { get; set; }
        public Site.Site Site { get; set; }

    }
}

File: DonationSite.Endpoint.API/ClientApp/DonationSite/src/models/donate.ts
import * as Interfaces from './../interfaces/donate';
export class Donate implements Interfaces.Donate {
  public donatorName: string = '';
  public value: number = 0;
  public fkSiteID: number = 0;
  constructor(siteId: number, donatorName: string, value: number) {
    this.fkSiteID = siteId;
    this.donatorName = donatorName;
    this.value = value;
  }
}
